ibu-imager/seedcleanup/seedcleanup.go

```diff
@@ -0,0 +1,224 @@
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/ops"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// Default location for etcdStaticPodFile
+	etcdStaticPodFile      = "/etc/kubernetes/manifests/etcd-pod.yaml"
+	etcdStaticPodContainer = "etcd"
+
+	containerListDoneFile = "/var/tmp/container_list.done"
+
+	// Default location for backing up seed cluster certificates
+	backupCertsDir = "/var/tmp/backupCertsDir"
+)
+
+// SeedCleaner handles cleanup operations after seeding, removing temporary files
+// and executing additional cleanup steps as needed.
+type SeedCleaner struct {
+	log                  *logrus.Logger
+	ops                  ops.Ops
+	backupDir            string
+	containerRegistry    string
+	authFile             string
+	recertContainerImage string
+	recertSkipValidation bool
+}
+
+// NewSeedCleaner is a constructor function for SeedCleaner
+func NewSeedCleaner(log *logrus.Logger, ops ops.Ops, backupDir,
+	containerRegistry, authFile, recertContainerImage string, recertSkipValidation bool) *SeedCleaner {
+
+	return &SeedCleaner{
+		log:                  log,
+		ops:                  ops,
+		backupDir:            backupDir,
+		containerRegistry:    containerRegistry,
+		authFile:             authFile,
+		recertContainerImage: recertContainerImage,
+		recertSkipValidation: recertSkipValidation,
+	}
+}
+
+// CleanupSeedCluster comprises the Imager workflow for cleanup operations after seeding an SNO cluster.
+func (s *SeedCleaner) CleanupSeedCluster() error {
+	s.log.Println("Cleaning up seed cluster")
+
+	// cleanup backup dir
+	if err := s.cleanupBackupDir(); err != nil {
+		return err
+	}
+
+	// cleanup preparation scripts and services
+	if err := s.cleanupConfigurationFiles(); err != nil {
+		return err
+	}
+
+	// restore certificates in etcd
+	if s.recertSkipValidation {
+		s.log.Info("Skipping restoring certificates in etcd via recert tool.")
+	} else {
+		if err := s.restoreCertificates(); err != nil {
+			return err
+		}
+	}
+
+	// restore cluster services
+	if err := s.restoreCluster(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SeedCleaner) cleanupBackupDir() error {
+	if err := os.RemoveAll(s.backupDir); err != nil {
+		return fmt.Errorf("error removing %s: %w", s.backupDir, err)
+	}
+
+	if err := os.Remove(containerListDoneFile); err != nil {
+		s.log.Errorf("error removing %s: %v", containerListDoneFile, err)
+	}
+
+	s.log.Info("Backup data removed successfully.")
+	return nil
+}
+
+func (s *SeedCleaner) cleanupConfigurationFiles() error {
+	systemdUnitsToDisable := []string{
+		"installation-configuration.service",
+		"prepare-installation-configuration.service",
+	}
+
+	for _, systemdUnit := range systemdUnitsToDisable {
+		if _, err := s.ops.SystemctlAction("disable", systemdUnit); err != nil {
+			s.log.Errorf("error disabling %s unit: %v", systemdUnit, err)
+		}
+	}
+	s.log.Info("Preparation systemd units disabled successfully.")
+
+	filesToRemove := []string{
```

Comment:
A general comment (inspired by this PR but not related to it directly) - I wonder if the imager should have all this knowledge about all those different services / scripts built into it, rather than receiving it from outside as parameters. I feel the imager is having a bit of a scope creep